### PR TITLE
JENKINS-46085 containerLog step to get the logs of a container running in the slave pod

### PR DESCRIPTION
**Pull request by marvinthepa**
Please don't merge yet, as I will add the missing parameters that the API supports
(`withTerminatedStatus sinceSeconds sinceTimestamp withTailingLines limitBytes`).

I just wanted to get early feedback, especially on the way I extracted the common code from `ContainerStepExecution`.

### CODE REVIEW

File: src/test/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/KubernetesPipelineTest.java [hunk 1]
```diff
@@ -321,6 +321,18 @@ public void declarative() throws Exception {
         r.assertLogContains("Apache Maven 3.3.9", b);
     }
 
+    @Issue("JENKINS-46085")
```
Comment (carlossg) [inline src/test/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/KubernetesPipelineTest.java]: can we have this test in a new file ContainerLogStepTest.java, this one is getting too big

**Comment by marvinthepa** [comment]
@carlossg: implemented the missing parameters, added license headers, extracted the test, fixed the build, and rebased master.

**Comment by marvinthepa** [comment]
Btw. this is https://issues.jenkins-ci.org/browse/JENKINS-46085

**Comment by carlossg** [comment]
I'm getting a test error

    mvn clean test -Dtest=KubernetesPipelineTest#runWithExistingEnvVariables
    ...
    Expected: a string containing "INSIDE_CONTAINER_ENV_VAR_LEGACY = container-env-var-value\n"

**Comment by marvinthepa** [comment]
Right, I made a mistake when rebasing master (too tired, probably).

Please try again.

**Comment by carlossg** [comment]
thanks!

**Comment by carlossg** [comment]
One thing that is missing is adding some docs to the readme, if you can add a new PR

**Pull request merged**